File: _archive/misc/fix_settings.py
import os
import re

main_xml = 'manager/src/main/res/xml/settings_main.xml'
dev_xml = 'manager/src/main/res/xml/settings_developer_options.xml'

# 1. Read settings_main.xml, extract category_dangerous_experimental and remove it
with open(main_xml, 'r') as f:
    main_content = f.read()

pattern = r'\s*<af\.shizuku\.manager\.settings\.CollapsiblePreferenceCategory[^>]+?key="category_dangerous_experimental".*?</af\.shizuku\.manager\.settings\.CollapsiblePreferenceCategory>'
match = re.search(pattern, main_content, flags=re.DOTALL)
if match:
    dangerous_block = match.group(0)
    main_content = main_content.replace(dangerous_block, '')
    with open(main_xml, 'w') as f:
        f.write(main_content)
    
    # 2. Add it to settings_developer_options.xml before </PreferenceScreen>
    with open(dev_xml, 'r') as f:
        dev_content = f.read()
    dev_content = dev_content.replace('</PreferenceScreen>', dangerous_block + '\n\n</PreferenceScreen>')
    with open(dev_xml, 'w') as f:
        f.write(dev_content)
    print("Moved category_dangerous_experimental to developer options.")

# 3. Remove all DiagramPreference and UiSettingsHeaderPreference blocks from xml
import glob
xml_files = glob.glob('manager/src/main/res/xml/*.xml')

pattern_diagram = r'\s*<af\.shizuku\.manager\.settings\.[a-zA-Z0-9_]*DiagramPreference[^>]*/>'
pattern_header = r'\s*<af\.shizuku\.manager\.settings\.UiSettingsHeaderPreference[^>]*/>'

for xml in xml_files:
    with open(xml, 'r') as f:
        content = f.read()
    
    new_content = re.sub(pattern_diagram, '', content, flags=re.DOTALL)
    new_content = re.sub(pattern_header, '', new_content, flags=re.DOTALL)
    
    if new_content != content:
        with open(xml, 'w') as f:
            f.write(new_content)
        print(f"Removed visuals from {xml}")

File: _archive/misc/fix_settings.js
const fs = require('fs');

const mainXml = 'manager/src/main/res/xml/settings_main.xml';
const devXml = 'manager/src/main/res/xml/settings_developer_options.xml';

let mainContent = fs.readFileSync(mainXml, 'utf8');

const pattern = /\s*<af\.shizuku\.manager\.settings\.CollapsiblePreferenceCategory[^>]+?key="category_dangerous_experimental"[\s\S]*?<\/af\.shizuku\.manager\.settings\.CollapsiblePreferenceCategory>/;
const match = mainContent.match(pattern);

if (match) {
    const dangerousBlock = match[0];
    mainContent = mainContent.replace(dangerousBlock, '');
    fs.writeFileSync(mainXml, mainContent);
    
    let devContent = fs.readFileSync(devXml, 'utf8');
    devContent = devContent.replace('</PreferenceScreen>', dangerousBlock + '\n\n</PreferenceScreen>');
    fs.writeFileSync(devXml, devContent);
    console.log("Moved category_dangerous_experimental to developer options.");
}

const glob = require('fs').readdirSync;
const xmlDir = 'manager/src/main/res/xml/';
const files = glob(xmlDir).filter(f => f.endsWith('.xml'));

const patternDiagram = /\s*<af\.shizuku\.manager\.settings\.[a-zA-Z0-9_]*DiagramPreference[^>]*\/>/g;
const patternHeader = /\s*<af\.shizuku\.manager\.settings\.UiSettingsHeaderPreference[^>]*\/>/g;

files.forEach(file => {
    const filePath = xmlDir + file;
    let content = fs.readFileSync(filePath, 'utf8');
    let newContent = content.replace(patternDiagram, '');
    newContent = newContent.replace(patternHeader, '');
    if (content !== newContent) {
        fs.writeFileSync(filePath, newContent);
        console.log(`Removed visuals from ${file}`);
    }
});
